src/libraries/System.Text.RegularExpressions/tests/RegexReductionTests.cs

```diff
@@ -339,29 +339,30 @@ private static int GetMinRequiredLength(Regex r)
         // Alternation reduction
         [InlineData("a|b", "[ab]")]
         [InlineData("a|b|c|d|e|g|h|z", "[a-eghz]")]
-        [InlineData("a|b|c|def|g|h", "[a-c]|def|[gh]")]
-        [InlineData("this|that|there|then|those", "th(?:is|at|ere|en|ose)")]
-        [InlineData("it's (?>this|that|there|then|those)", "it's (?>th(?:is|at|ere|en|ose))")]
-        [InlineData("it's (?>this|that|there|then|those)!", "it's (?>th(?:is|at|ere|en|ose))!")]
+        [InlineData("a|b|c|def|g|h", "(?>[a-c]|def|[gh])")]
+        [InlineData("this|that|there|then|those", "th(?>is|at|ere|en|ose)")]
+        [InlineData("it's (?>this|that|there|then|those)", "it's (?>th(?>is|at|e(?>re|n)|ose))")]
+        [InlineData("it's (?>this|that|there|then|those)!", "it's (?>th(?>is|at|e(?>re|n)|ose))!")]
         [InlineData("abcd|abce", "abc[de]")]
-        [InlineData("abcd|abef", "ab(?:cd|ef)")]
-        [InlineData("abcd|aefg", "a(?:bcd|efg)")]
-        [InlineData("abcd|abc|ab|a", "a(?:bcd|bc|b|)")]
-        [InlineData("abcde|abcdef", "abcde(?:|f)")]
-        [InlineData("abcdef|abcde", "abcde(?:f|)")]
+        [InlineData("abcd|abef", "ab(?>cd|ef)")]
+        [InlineData("abcd|aefg", "a(?>bcd|efg)")]
+        [InlineData("abcd|abc|ab|a", "a(?>bcd|bc|b|)")]
+        [InlineData("abcde|abcdef", "abcde(?>|f)")]
+        [InlineData("abcdef|abcde", "abcde(?>f|)")]
         [InlineData("abcdef|abcdeg|abcdeh|abcdei|abcdej|abcdek|abcdel", "abcde[f-l]")]
         [InlineData("(ab|ab*)bc", "(a(?:b|b*))bc")]
         [InlineData("abc(?:defgh|defij)klmn", "abcdef(?:gh|ij)klmn")]
         [InlineData("abc(defgh|defij)klmn", "abc(def(?:gh|ij))klmn")]
         [InlineData("a[b-f]|a[g-k]", "a[b-k]")]
         [InlineData("this|this", "this")]
         [InlineData("this|this|this", "this")]
-        [InlineData("hello there|hello again|hello|hello|hello|hello", "hello(?: there| again|)")]
-        [InlineData("hello there|hello again|hello|hello|hello|hello|hello world", "hello(?: there| again|| world)")]
-        [InlineData("hello there|hello again|hello|hello|hello|hello|hello world|hello", "hello(?: there| again|| world)")]
+        [InlineData("hello there|hello again|hello|hello|hello|hello", "hello(?> there| again|)")]
```

Comment:
Does it mean that the old (non-atomic) left and right patterns now reduce to same code (and hence fail `PatternsReduceDifferently` test)?

Reply:
The previous version of the test passes before and after this change.  But with this change,  the alternation becomes atomic, and this test now validates that as well; the tweaked variant only passes after this PR.

Reply:
Ah, I missed the collapsed method name (`PatternsReduceIdentically`) in diff view and thought it's the attribute on `PatternsReduceDifferently`. Thanks for the explanation. 🙂